services/storage/src/simcore_service_storage/access_layer.py

```diff
@@ -0,0 +1,266 @@
+""" Helper functions to determin access-rights on stored data
+
+# DRAFT Rationale:
+
+    osparc-simcore defines TWO authorization methods: i.e. a set of rules on what,
+    how and when any resource can be accessed or operated by a user
+
+    ## ROLE-BASED METHOD:
+        In this method, a user is assigned a role (user/tester/admin) upon registration. Each role is
+        system-wide and defines a set of operations that the user *can* perform
+            - Every operation is named as a resource and an action (e.g. )
+            - Resource is named hierarchically
+            - Roles can inherit permitted operations from other role
+        This method is static because is system-wide and it is defined directly in the
+        code at services/web/server/src/simcore_service_webserver/security_roles.py
+        It is defined on top of every API entrypoint and applied just after authentication of the user.
+
+    ## GROUP-BASED METHOD:
+        The second method is designed to dynamically share a resource among groups of users. A group
+        defines a set of rules that apply to a resource and users can be added to the group dynamically.
+        So far, there are two resources that define access rights (AR):
+            - one applies to projects (read/write/delete) and
+            - the other to services (execute/write)
+        The project access rights are set in the column "access_rights" of the "projects" table .
+        The service access rights has its own table: service_access_rights
+
+        Access rights apply hierarchically, meaning that the access granted to a project applies
+        to all nodes inside and stored data in nodes.
+
+        How do these two AR coexist?: Access to read, write or delete a project are defined in the project AR but execution
+        will depend on the service AR attached to nodes inside.
+
+        What about stored data?
+        - data generated in nodes inherits the AR from the associated project
+        - data generated in API uses full AR provided by ownership (i.e. user_id in files_meta_data table)
```

Comment:
I am not sure I do understand the inherent difference between the API and the others..
the nodes always generate data owned by the current user (be it from the WebAPI or the API) right? we do not differentiate at the moment data generated as output of a full pipeline and data generated by just a node inside a pipeline. maybe you could clarify?

Reply:
the "api" data refers to data uploaded via the public API. All entrypoints there are authenticated and therefore the data is directly owned by the user. It should not be seen as neither input/output data but just user's data.

Reply:
I agree with Pedro @pcrespov here